proto/data.go

```diff
@@ -330,19 +329,34 @@ func (v *Value) SetInteger(i int64) {
 	v.Bytes = encoding.EncodeUint64(nil, uint64(i))
 }
 
-// GetInteger decodes an int64 value from the bytes field of the receiver.
-func (v *Value) GetInteger() int64 {
-	if v == nil || v.Bytes == nil {
-		return 0
+// GetInteger decodes an int64 value from the bytes field of the receiver. If
+// the bytes field is not 0 or 8 bytes in length an error will be returned.
+func (v *Value) GetInteger() (int64, error) {
+	if v == nil || len(v.Bytes) == 0 {
+		return 0, nil
+	}
+	const uint64Size = int(unsafe.Sizeof(uint64(0)))
```

Comment:
`{Encode,Decode}Uint64` are defined to be fixed-width encodings. Variable width encodings are available via the `{Encode,Decode}Varint` routines which I agree should have a constant associated with their max encoded length if that length is needed.